### ISSUE TITLE
[css-ui-4] `input-security` may be misleading

### ISSUE THREAD

**Issue by april** [w3c/csswg-drafts#6449]
Hey there! Recently stumbled across #6239, as I was asking around to see if browsers were working on a way to standardize the "bisected-eye" password reveal functionality. Edge always has it, Chrome sometimes does, Firefox doesn't, and it's a bit of an a11y nightmare as a result.

Anyways, I was quite excited to see the `input-security` proposal in #6239. Wonderful that there has been some progress on this issue. That said, I have a concern I'd like to share, from my perspective as a former security engineer at Mozilla.

The name `input-security` is fairly vague in what it is offering, and may provide people with a false sense of security as to what it does (especially as `auto`). It's clear if you read the specification, but in a general sense it may mislead people e.g. does `input-security` prevent Javascript from reading the contents or not.

What I'd like to propose is something a bit more direct in its wording, `text-concealment-toggle`:

• `text-concealment-toggle: visible` (shows the bisected eye toggle)
• `text-concealment-toggle: none` (no bisected eye)
• `text-concealment-toggle: auto` (lets the user agent decide)

With the possible suggestion that user agents SHOULD show the toggle with `auto`, for accessibility reasons.

This would also allow a more general-purpose attribute for control over the concealment of input in general, if wanted, such as on `<input type="tel">`:

• `text-concealment: auto` (lets the user agent decide)
• `text-concealment: concealed` (shows asterisks)
• `text-concealment: none` (never shows asterisks)

Thanks so much!

**Comment by SebastianZ** [comment]
I agree with your concern that `input-security` can be misleading, though I'm not happy with the proposed name. While it covers its functionality, "concealment" is long, hard to spell and also hard to understand for non-English speakers.

Sebastian

**Comment by jonjohnjohnson** [comment]
Or maybe one of these?
- `input-conceal`
- `input-pretext`
- `input-disguise`
- `input-obscure`

**Comment by april** [comment]
Yeah, I agree that it’s not ideal, there just aren’t a lot of words that mean the same thing in English. `text-conceal` (or `input-conceal`) is shorter, seems like it could work. `obscure` is slightly less descriptive (imo) but I think it also gets the point across.

**Comment by SebastianZ** [comment]
From @jonjohnjohnson's list I'd prefer `input-obscure`. As I said before, `conceal` is rather hard to spell and presumably also less known to non-English people. Also `disguise` is quite hard to spell and `pretext` doesn't really fit the purpose, in my opinion.

Having said that, what speaks against a simple `input-display` or `input-text-display` with the values `auto` (UA decides how to display the text), `obscure` (display the text concealed), and `normal` (display the text as is)?

Sebastian

**Comment by april** [comment]
`conceal` is a quite common word in the English language, and is considered by pretty much all lexical analyzers to be a reasonably low-complexity English word. Both `conceal` and `concealment` are used more in English than `obscure` is, and by a fairly considerable margin in both ngram analysis as well as Google search results.

It is far, far more common than words already in CSS such as "ligature" or even "opacity". While I think "obscure" is a fine choice, as a word it sees less use than "conceal" in English. That said, "obscure" does have more similar-sounding words in romance languages. I think either would be fine.

As for `display` or `input-display`, `display` already has a meaning in CSS, and it covers whether or not to show/render something. Password fields do show something, it's just concealed by dots or asterisks.

**Comment by SebastianZ** [comment]
You have a point there regarding `conceal`.

My intention suggesting `input-display` was to make the property name more general and put the control of how the input text is displayed to the value. Doing so makes the property a little more extensible for new display types. It could also be called `input-appearance`, `input-rendering`, or something similar. It's just an idea, though, and I don't have a strong opinion about it.

Sebastian
